internal/ir/xds.go

```diff
@@ -2251,3 +2253,20 @@ type DestinationFilters struct {
 	// RemoveResponseHeaders defines a list of headers to be removed from response.
 	RemoveResponseHeaders []string `json:"removeResponseHeaders,omitempty" yaml:"removeResponseHeaders,omitempty"`
 }
+
+// ResourceMetadata is metadata from the provider resource that is translated to an envoy resource
+// +k8s:deepcopy-gen=true
+type ResourceMetadata struct {
```

Comment:
thoughts on leaving out `SectionName` and `GroupVersion` in the first iteration ?

Reply:
will remove `GroupVersion`. I'll add VHost to this PR in a bit, and it'll include a useful `SectionName`, so prefer to keep it.